apps/spruce/src/pages/waterfall/NameFilter.tsx

```diff
@@ -0,0 +1,41 @@
+import { useCallback } from "react";
+import { useWaterfallAnalytics } from "analytics";
+import TupleSelectWithRegexConditional from "components/TupleSelectWithRegexConditional";
+import { useUpsertQueryParams } from "hooks";
+import { WaterfallFilterOptions } from "types/waterfall";
+import { validators } from "utils";
+
+export const NameFilter = () => {
+  const onSubmit = useUpsertQueryParams();
+  const { sendEvent } = useWaterfallAnalytics();
+
+  const onSubmitTupleSelect = useCallback(
+    ({ category, value }: { category: string; value: string }) => {
+      onSubmit({ category, value });
+      switch (category) {
+        case WaterfallFilterOptions.BuildVariant:
+          sendEvent({ name: "Created build variant filter" });
+          break;
+        default:
+      }
+    },
+    [onSubmit, sendEvent],
+  );
+
+  return (
+    <TupleSelectWithRegexConditional
+      onSubmit={onSubmitTupleSelect}
```

Comment:
Remind me how exact vs. regex is distinguished in the query params again? I haven't been able to observe a difference in behavior

Reply:
If exact is selected, the string will be escaped in the query param.